src/modules/power_profiles_daemon.cpp

```diff
@@ -0,0 +1,147 @@
+#include "modules/power_profiles_daemon.hpp"
+
+// In the 80000 version of fmt library authors decided to optimize imports
+// and moved declarations required for fmt::dynamic_format_arg_store in new
+// header fmt/args.h
+#if (FMT_VERSION >= 80000)
+#include <fmt/args.h>
+#else
+#include <fmt/core.h>
+#endif
+
+#include <spdlog/spdlog.h>
+#include <glibmm.h>
+#include <glibmm/variant.h>
+
+
+
+namespace waybar::modules {
+
+PowerProfilesDaemon::PowerProfilesDaemon(const std::string& id, const Json::Value& config)
+  : ALabel(config, "power-profiles-daemon", id, "{profile}", 0, false, true)
+{
+  // NOTE: the DBus adresses are under migration. They should be
+  // changed to org.freedesktop.UPower.PowerProfiles at some point.
+  //
+  // See
+  // https://gitlab.freedesktop.org/upower/power-profiles-daemon/-/releases/0.20
+  //
+  // The old name is still announced for now. Let's rather use the old
+  // adresses for compatibility sake.
+  //
+  // Revisit this in 2026, systems should be updated by then.
+  power_profiles_proxy_ = Gio::DBus::Proxy::create_for_bus_sync(Gio::DBus::BusType::BUS_TYPE_SYSTEM,
+                                                        "net.hadess.PowerProfiles", "/net/hadess/PowerProfiles",
+                                                        "net.hadess.PowerProfiles");
+  if (!power_profiles_proxy_) {
+    spdlog::error("PowerProfilesDaemon: DBus error, cannot connect to net.hasdess.PowerProfile");
+  } else {
+    // Connect active profile callback
+    powerProfileChangeSignal_ = power_profiles_proxy_->signal_properties_changed()
+      .connect(sigc::mem_fun(*this, &PowerProfilesDaemon::profileChanged_cb));
+    populateInitState();
+    dp.emit();
+  }
+}
+
+// Look for the profile str in our internal profiles list. Using a
+// vector to store the profiles ain't the smartest move
+// complexity-wise, but it makes toggling between the mode easy. This
+// vector is 3 elements max, we'll be fine :P
+void PowerProfilesDaemon::switchToProfile_(std::string str) {
+  auto pred = [str](Profile p) { return p.name == str; };
+  activeProfile_ = std::find_if(availableProfiles_.begin(), availableProfiles_.end(), pred);
+  if (activeProfile_ == availableProfiles_.end()) {
+    spdlog::error("PowerProfilesDaemon: FATAL, can't find the active profile in the available profiles list");
+    exit(-1);
```

Comment:
why an exit?

Reply:
Erf! Two reasons 😅

1. When I wrote this, I did not know the best way to crash the module to prevent it to do some dangerous dereferences. I made a mental note about that and put this exit in the meantime.
2. My memory is not so good, I tend to forget my mental notes very fast 😀

Anyway, after looking the current codebase, it seems like throwing a runtime error does the trick. Don't hesitate to point me to another direction if this is not the best way to go.